tools/duplicate_function_eliminator.py

```diff
@@ -247,8 +247,11 @@ def write_chunk(chunk, i):
       if DEBUG: print >> sys.stderr, 'splitting up js optimization into %d chunks, using %d cores  (total: %.2f MB)' % (len(chunks), cores, total_size/(1024*1024.))
       pool = multiprocessing.Pool(processes=cores)
       filenames = pool.map(run_on_chunk, commands, chunksize=1)
-      pool.terminate()
-      pool.join()
+      try:
+        pool.terminate()
+        pool.join()
+      except Exception, e:
```

Comment:
Oh sorry, my bad in explaining here. This PR did not cause those errors, but those have been always occurring with low frequency, on certain machines. The new Windows bot runs into those quite often, see the latest Windows bot run for example: http://clb.demon.fi:8112/builders/win-incoming/builds/1284/steps/default%20suite/logs/stdio

and search for "Access is denied". One such dump:

    test_dlfcn_unique_sig (test_core.asm3) ... Traceback (most recent call last):
      File "C:\slaves\emslave\buildslave\win-incoming\emsdk\emscripten\incoming\emcc", line 13, in <module>
        emcc.run()
      File "C:\slaves\emslave\buildslave\win-incoming\emsdk\emscripten\incoming\emcc.py", line 1784, in run
        JSOptimizer.flush()
      File "C:\slaves\emslave\buildslave\win-incoming\emsdk\emscripten\incoming\emcc.py", line 1681, in flush
        run_passes(chunks[0], title, just_split=False, just_concat=False)
      File "C:\slaves\emslave\buildslave\win-incoming\emsdk\emscripten\incoming\emcc.py", line 1654, in run_passes
        final = shared.Building.js_optimizer(final, passes, debug_level >= 4, JSOptimizer.extra_info, just_split=just_split, just_concat=just_concat)
      File "C:\slaves\emslave\buildslave\win-incoming\emsdk\emscripten\incoming\tools\shared.py", line 1748, in js_optimizer
        ret = js_optimizer.run(filename, passes, NODE_JS, debug, extra_info, just_split, just_concat)
      File "C:\slaves\emslave\buildslave\win-incoming\emsdk\emscripten\incoming\tools\js_optimizer.py", line 544, in run
        return temp_files.run_and_clean(lambda: run_on_js(filename, passes, js_engine, source_map, extra_info, just_split, just_concat))
      File "C:\slaves\emslave\buildslave\win-incoming\emsdk\emscripten\incoming\tools\tempfiles.py", line 64, in run_and_clean
        return func()
      File "C:\slaves\emslave\buildslave\win-incoming\emsdk\emscripten\incoming\tools\js_optimizer.py", line 544, in <lambda>
        return temp_files.run_and_clean(lambda: run_on_js(filename, passes, js_engine, source_map, extra_info, just_split, just_concat))
      File "C:\slaves\emslave\buildslave\win-incoming\emsdk\emscripten\incoming\tools\js_optimizer.py", line 447, in run_on_js
        pool.terminate()
      File "C:\slaves\emslave\buildslave\win-incoming\emsdk\python\2.7.5.3_64bit\lib\multiprocessing\pool.py", line 452, in terminate
        self._terminate()
      File "C:\slaves\emslave\buildslave\win-incoming\emsdk\python\2.7.5.3_64bit\lib\multiprocessing\util.py", line 201, in __call__
        res = self._callback(*self._args, **self._kwargs)
      File "C:\slaves\emslave\buildslave\win-incoming\emsdk\python\2.7.5.3_64bit\lib\multiprocessing\pool.py", line 500, in _terminate_pool
        p.terminate()
      File "C:\slaves\emslave\buildslave\win-incoming\emsdk\python\2.7.5.3_64bit\lib\multiprocessing\process.py", line 137, in terminate
        self._popen.terminate()
      File "C:\slaves\emslave\buildslave\win-incoming\emsdk\python\2.7.5.3_64bit\lib\multiprocessing\forking.py", line 312, in terminate
        _subprocess.TerminateProcess(int(self._handle), TERMINATE)
    WindowsError: [Error 5] Access is denied
    Error in atexit._run_exitfuncs:
    Traceback (most recent call last):
      File "C:\slaves\emslave\buildslave\win-incoming\emsdk\python\2.7.5.3_64bit\lib\atexit.py", line 24, in _run_exitfuncs
        func(*targs, **kargs)
      File "C:\slaves\emslave\buildslave\win-incoming\emsdk\python\2.7.5.3_64bit\lib\multiprocessing\util.py", line 315, in _exit_function
        p._popen.terminate()
      File "C:\slaves\emslave\buildslave\win-incoming\emsdk\python\2.7.5.3_64bit\lib\multiprocessing\forking.py", line 312, in terminate
        _subprocess.TerminateProcess(int(self._handle), TERMINATE)
    WindowsError: [Error 5] Access is denied
    Error in sys.exitfunc:
    Traceback (most recent call last):
      File "C:\slaves\emslave\buildslave\win-incoming\emsdk\python\2.7.5.3_64bit\lib\atexit.py", line 24, in _run_exitfuncs
        func(*targs, **kargs)
      File "C:\slaves\emslave\buildslave\win-incoming\emsdk\python\2.7.5.3_64bit\lib\multiprocessing\util.py", line 315, in _exit_function
        p._popen.terminate()
      File "C:\slaves\emslave\buildslave\win-incoming\emsdk\python\2.7.5.3_64bit\lib\multiprocessing\forking.py", line 312, in terminate
        _subprocess.TerminateProcess(int(self._handle), TERMINATE)
    WindowsError: [Error 5] Access is denied
    FAIL

The root cause is the same as what is causing bug #718, but in that bug, the discussion related to a different code site where it stemmed from.

I could have made the catch only grab `WindowsError`s instead of all errors, but seems like if we get any error in this cleanup, the only sensible thing we can do is to just go on and ignore.